src/process_args.c

```diff
@@ -175,6 +175,15 @@ process_args(nmsgtool_ctx *c) {
 			c->kicker = strdup(t);
 	}
 
+#if defined(HAVE_LIBRDKAFKA) || defined(HAVE_JSON_C)
+	/* kafka key */
+	if (c->kafka_key_field == NULL) {
+		t = getenv("NMSG_KAFKA_KEY");
+		if (t != NULL)
+			c->kafka_key_field = strdup(t);
```

Comment:
Might make more sense to call strdup() inside `nmsg_output_open_kafka_json()` rather than here (and also free() it at `nmsg_output_close()`)